edgelet/edgelet-http-hosting/src/client/hosting.rs

```diff
@@ -0,0 +1,219 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use std::sync::Arc;
+
+use failure::{Fail, ResultExt};
+use futures::prelude::*;
+use hosting::apis::client::APIClient;
+use hosting::apis::configuration::Configuration;
+use hosting::apis::HostingApi;
+use hosting::models::*;
+use hyper::Client;
+use url::Url;
+
+use edgelet_core::UrlExt;
+use edgelet_http::UrlConnector;
+
+use crate::error::{Error, ErrorKind};
+
+pub trait HostingInterface {
```

Comment:
Nit: Let's call this provisioning, instead of hosting

Reply:
Should I rename it to external-provisioning/ExternalProvisoning instead? Just to be clear about the fact that all of this is related to just the external provisioning mode?

Reply:
Should I rename it to external-provisioning/ExternalProvisoning instead? Just to be clear about the fact that all of this is related to just the external provisioning mode?

---
In reply to: [282089902](https://github.com/Azure/iotedge/pull/1144#discussion_r282089902) [](ancestors = 282089902)

Reply:
Yes, I agree with `external-provisioning` over `hosting` / `provisioning`.

Reply:
Cool, gonna go with external-provisioning